build/v1/consts.go

```diff
@@ -0,0 +1,126 @@
+	// not completed and retrying the build times out.
+	StatusReasonExceededRetryTimeout StatusReason = "ExceededRetryTimeout"
+
+	// StatusReasonMissingPushSecret indicates that the build is missing required
+	// secret for pushing the output image.
+	// The build will stay in the pending state until the secret is created, or the build times out.
+	StatusReasonMissingPushSecret StatusReason = "MissingPushSecret"
+
+	// StatusReasonPostCommitHookFailed indicates the post-commit hook failed.
+	StatusReasonPostCommitHookFailed StatusReason = "PostCommitHookFailed"
+
+	// StatusReasonPushImageToRegistryFailed indicates that an image failed to be
+	// pushed to the registry.
+	StatusReasonPushImageToRegistryFailed StatusReason = "PushImageToRegistryFailed"
+
+	// StatusReasonPullBuilderImageFailed indicates that we failed to pull the
+	// builder image.
+	StatusReasonPullBuilderImageFailed StatusReason = "PullBuilderImageFailed"
+
+	// StatusReasonFetchSourceFailed indicates that fetching the source of the
+	// build has failed.
+	StatusReasonFetchSourceFailed StatusReason = "FetchSourceFailed"
+
+	// StatusReasonInvalidContextDirectory indicates that the supplied
+	// contextDir does not exist
+	StatusReasonInvalidContextDirectory StatusReason = "InvalidContextDirectory"
+
+	// StatusReasonCancelledBuild indicates that the build was cancelled by the
+	// user.
+	StatusReasonCancelledBuild StatusReason = "CancelledBuild"
+
+	// StatusReasonDockerBuildFailed indicates that the docker build strategy has
+	// failed.
+	StatusReasonDockerBuildFailed StatusReason = "DockerBuildFailed"
+
+	// StatusReasonBuildPodExists indicates that the build tried to create a
+	// build pod but one was already present.
+	StatusReasonBuildPodExists StatusReason = "BuildPodExists"
+
+	// StatusReasonNoBuildContainerStatus indicates that the build failed because the
+	// the build pod has no container statuses.
+	StatusReasonNoBuildContainerStatus StatusReason = "NoBuildContainerStatus"
+
+	// StatusReasonFailedContainer indicates that the pod for the build has at least
+	// one container with a non-zero exit status.
+	StatusReasonFailedContainer StatusReason = "FailedContainer"
+
+	// StatusReasonUnresolvableEnvironmentVariable indicates that an error occurred processing
+	// the supplied options for environment variables in the build strategy environment
+	StatusReasonUnresolvableEnvironmentVariable StatusReason = "UnresolvableEnvironmentVariable"
+
+	// StatusReasonGenericBuildFailed is the reason associated with a broad
+	// range of build failures.
+	StatusReasonGenericBuildFailed StatusReason = "GenericBuildFailed"
+
+	// StatusCannotRetrieveServiceAccount is the reason associated with a failure
+	// to look up the service account associated with the BuildConfig.
+	StatusReasonCannotRetrieveServiceAccount StatusReason = "CannotRetrieveServiceAccount"
+
+	StatusMessageCannotCreateBuildPodSpec        = "Failed to create pod spec."
```

Comment:
These are all human readable strings that are part of status, but they aren't part of the guaranteed status API.  I don't think I'd put the in openshift/api because anyone referencing them could be broken at any time.  I think it is ok for these to change over time and I think my response to anyone complaining would be, "these are human readable niceties, but not guaranteed API"

As I understand the other side from @bparees, both build controllers and builder pods update status messages and it is nice for them to use the same human readable string.  Drift would be hard to manage.  @bparees please fill in additional detail.

@smarterclayton @liggitt This is likely to be the first of many, where shall we start?

Reply:
In general i'm just looking for a home for constants with an eye to getting rid of dependencies on origin.  Right now, openshift/api is the best "common" location where I can put them. 

My concerns about not putting them here are:
1) drift if we ended up needing to make multiple copies as @deads2k noted
2) these messages are strongly related to the other constants (StatusReason fields) so i don't really want to separate them, from a code management perspective.  (Moving all these constants elsewhere would be acceptable but again that elsewhere is not currently defined, short of "make a copy of them in each place you need to use them")

Reply:
I don't feel like being blocked on this so i'm just going to move them as requested.